applications/crossbar/src/api_util.erl

```diff
@@ -868,17 +868,24 @@ ensure_content_type(CT) -> CT.
 %% (the final module in the chain) accepts this verb parameter pair.
 %% @end
 %%--------------------------------------------------------------------
--spec does_resource_exist(cb_context:context()) -> boolean().
--spec does_resource_exist(cb_context:context(), list()) -> boolean().
+-spec does_resource_exist(cb_context:context()) -> {boolean(), cb_context:context()}.
+-spec does_resource_exist(cb_context:context(), list()) -> {boolean(), cb_context:context()}.
 does_resource_exist(Context) ->
     does_resource_exist(Context, cb_context:req_nouns(Context)).
 
 does_resource_exist(Context, [{Mod, Params}|_]) ->
     Event = api_util:create_event_name(Context, <<"resource_exists.", Mod/binary>>),
-    Responses = crossbar_bindings:map(Event, Params),
-    crossbar_bindings:any(Responses) and 'true';
-does_resource_exist(_Context, _ReqNouns) ->
-    'false'.
+    Responses = crossbar_bindings:map(Event, [Context, Params]),
```

Comment:
@jamesaimonetti 
Sorry, haven't enough time to answer.

I called `crossbar_bindings:map` in this way to have only one function with arity 2.
It prevents spagetti of specs.
Developer can add clause to match "generic" situation.
Developer still can pattern match all needed variables.
In my opinion `resource_exists` should check if resource realy exists, and can skip checks for actions. Actions may be checked in `allowed_methods`.
Getting `/v../accounts/.../<resource_type>/<resource_id>` and `/v../accounts/.../<resource_type>/<resource_id>/<action>` will check the same resource. Do we realy need several function for this?